rclpy/src/rclpy/graph.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLPY__GRAPH_HPP_
+#define RCLPY__GRAPH_HPP_
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/graph.h>
+
+#include <memory>
+#include <string>
+
+namespace py = pybind11;
+
+namespace rclpy
+{
+
+/// Get topic names and types for which a given remote node has publishers.
+/**
+ * Raises ValueError if pynode is not a node capsule
+ * Raises RCLError if there is an rcl error
```

Comment:
Please also add `NodeNameNonExistentError`

Reply:
I was going to ask this. `NodeNameNonExistant` extends `RCLError`. Should it be separately mentioned? If so, then all of the docstrings need to be updated.

Reply:
Done in https://github.com/ros2/rclpy/pull/717/commits/a13f583864c6a3ff3d68fbd577acb3e86361eb07.